docker-buildscript.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+
+build_docker_image() {
+    # local BASE_IMAGE=$1
+    # local TARGET=$2
+    # docker pull ${BASE_IMAGE}
+    local BASE_DIGEST=$(docker inspect --format='{{index .RepoDigests 0}}' ${BASE_IMAGE})
+    # Extract the digest part (after the @ symbol)
+    local BASE_DIGEST=${BASE_DIGEST#*@}
+
+    
+
+    local VCS_REF=$(git rev-parse HEAD)
+    local VSC_SHORT_REF=$(git rev-parse --short HEAD)
+
+    local IMAGE_TAG="ghcr.io/utilacy/${TARGET}:commit-${VSC_SHORT_REF}"
+
+    docker build \
+        --target "$TARGET" \
+        --build-arg BASE_IMAGE=${BASE_IMAGE} \
+        --build-arg BASE_DIGEST=${BASE_DIGEST} \
+        --build-arg BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ") \
+        --build-arg VCS_REF=$(git rev-parse --short HEAD) \
+        --build-arg DESCRIPTION="${DESCRIPTION}" \
+        --build-arg REF_NAME=${IMAGE_TAG} \
+        -t ${IMAGE_TAG} \
+        .
+
+    if [ "$TAG_AS_RELEASE" = true ]; then
+        # Extract the version of the specified TARGET package from Cargo metadata using jq
+        local TARGET_VERSION=$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[] | select(.name=="'${TARGET}'")| .version')
+        local RELEASE_TAG="ghcr.io/utilacy/${TARGET}:${TARGET_VERSION}"
+        echo "Tagging image as ${RELEASE_TAG}"
+        docker tag ${IMAGE_TAG} ${RELEASE_TAG}
+    fi
+
+    if [ "$TAG_AS_LATEST" = true ]; then
+        local LATEST_TAG="ghcr.io/utilacy/${TARGET}:latest"
+        echo "Tagging image as ${LATEST_TAG}"
+        docker tag ${IMAGE_TAG} ${LATEST_TAG}
+    fi
+}
+
+# Set the default base image if it is not set
+: "${BASE_IMAGE:="debian:bookworm-slim"}"
+
+TARGET=rosenpass
+DESCRIPTION="The reference implementation of the protocol - the rosenpass tool"
+build_docker_image
+
+TARGET=rp
+DESCRIPTION="A frontend integrating Rosenpass and WireGuard to create a vpn - the rp frontend"
```

Comment:
```suggestion
DESCRIPTION="A frontend integrating Rosenpass and WireGuard to create a VPN - the rp frontend"
```

Reply:
Alternative suggestions/phrasing:

```suggestion
DESCRIPTION="The rp frontend integrates Rosenpass and WireGuard to create a VPN"
```

```suggestion
DESCRIPTION="The rp command-line integrates Rosenpass and WireGuard to help you create a VPN"
```

```suggestion
DESCRIPTION="Command-line application integrating Rosenpass and WireGuard to create a VPN - the rp frontend"
```

```suggestion
DESCRIPTION="rp is a frontend integrating Rosenpass and WireGuard to help you create a VPN"
```

(Or some variation thereof. You could also consider replacing `create` with `set up` or even `manage`?)

Reply:
I removed the buildscript